### ISSUE TITLE
Transition Model which Attaches the Transition Object to the clips so that the Transition is not broken on following edits

### ISSUE THREAD

**Issue by Twenkid** [OpenShot/openshot-qt#3082]
**Describe the new feature:**
Modified way of working with Transitions, e.g. one which attaches the Transition Object to one of the clips, so the transition is not broken on each new edit and when the clip is extended or shortened, the transition adapts accordingly. That would also save the manual re-creation of transitions following the current one by dragging.

When moving clips around, sometimes it's possible to miss to select a transition, to move it slightly more or less and sometimes that is noticed only after rendering as wrong frames (the proposed features of condensing etc. by a command may help to reduce the latter cases).

**Describe the solution you'd like:**
The Transition objects to be attached to one (or both) of the clips, so when moving or editing the clips, the transition is moved and rendered accordingly and is not broken (representing not the right end/beginning of the adjacent clips).

**Describe alternatives you have considered:**
Currently when some of the clips with transitions is edited and some segment has to be moved around, that may involve regenerating all of the following transitions by manually dragging partial coverage etc. - laborious and not precise.

**Comment by SuslikV** [comment]
Probably, it can be complete as some property that links objects position during drag operation to some clip position (each clip has three time properties: "position", "start", "end" - that can be adjusted).

Just my thoughts. I have no plans to implement this.

**Comment by ferdnyc** [comment]
Yeah, it's funny how things ended up falling out, with these decisions... 

* **Effects** ended up getting attached to a clip, when they don't _need_ to be (in terms of `libopenshot`) and it would actually be kind of handy if they **weren't**.
    (So that you could apply an effect across several clips... or have it start and stop somewhere other than at a clip boundary.)
* **Transitions**, OTOH, ended up getting treated as completely independent objects, when really they don't make any _sense_ unless they're attached to at least one clip.
   (You can't use a transition in the middle of a clip, after all, since it won't do the right thing at all. Transitions only make sense at the start or end of a clip. Heck, it really shouldn't even be _possible_ to place them anywhere else.)

As a result, we have feature requests in to make effects copy/pastable between clips (which they shouldn't be attached to), _and_ we have feature requests in to prevent editing from shifting transitions off of the edges of clips (which _they_ **should** be attached to). \*sigh\*

**Comment by Twenkid** [comment]
Hello, as of the effects - I think it's reasonable to have both kinds of placements - two layers of effects, current and independent of the clips as ferdnyc's suggested. I think current is needed, because during editing it is expected to have clips changing their lengths and positions, so if an effect is independent, at least some constraints may be added, such as application between markers, or a choice to attach it to particular clips, which may be not adjacent, but a set of clips; also a set of interval-selections.

BTW, the second type, clip-independent, was also a "feature request" from myself-to-myself in an in-house video editor, which also first had effects attached to clips, but there should be another layer also, which still lacks like in OpenShot... :)
